qemu/tests/virtio_trace_pipenb.py

```diff
@@ -0,0 +1,42 @@
+import time
+from aexpect import ShellStatusError, ShellTimeoutError
+
+from avocado.utils import process
+from virttest import error_context
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Under named-pipe non-blocking testing:
+    1) Create pipe named by the following
+    2) Boot up a single-CPU guest with a virtio-serial device and
+       named-pipe chardev backend
+    3) Run a background work on the guest.
+    4) Write data to the virtio-serial port until the guest stops.
+    5) Read the named-pipe file on the host.
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+
+    timeout = float(params.get("login_timeout", 360))
+    vm = env.get_vm(params["main_vm"])
+    serials = params["serials"].split()
+    v_path = vm.get_serial_console_filename(serials[-1])
+    vm.verify_alive()
+    session = vm.wait_for_login(timeout=timeout)
+    assert session.cmd_status_output(" while :;do sleep 1; date; done &") is not None, "There should be date in the output"
+    session.cmd_status_output("cat /proc/kallsyms > /dev/virtio-ports/vs2 &")
+    time.sleep(10)
+    out_put = process.getoutput("cat {}".format(v_path), timeout=timeout, shell=True)
+    assert 'compile_date_show' in out_put, "There should be date in the output"
+    assert 'Done' in session.cmd('ps -ef | grep "cat /proc/kallsyms > /dev/virtio-ports/vs2 &"', timeout=timeout), 'Cat command is not finished'
+    try:
+        assert session.cmd('ps -aux | grep "while :;do sleep 1; date; done &"', timeout=timeout) is not None
```

Comment:
This command will not exit automatically. Could you explain why we need check this ?

Reply:
Hi nanliu want to make sure this command still running...

Reply:
> This command will not exit automatically. Could you explain why we need check this ?

I want to check whether there still have output after the data transfer.